### PR TITLE
AA - Create event page

### PR DESCRIPTION
**Pull request by AntonovAtanas**
- add create event page with stylization
- add camera, filesystem, and pwa dependency for testing

- TODO: Handle form submittion & validation
- TODO: Add proper categories enum and use modal on ion-selects
- TODO: Pick date calendar

image
image
image
image
image
image

### CODE REVIEW

File: src/app/events/event-create/event-create.page.html [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+  <app-header
+    [headerTitle]="headerTitle"
+    [backButton]="backButton"
+    [defaultHref]="defaultHref"
+  ></app-header>
+
+  <div class="create-event-container">
+    <form
+      #createEventForm="ngForm"
+      (submit)="onCreateEventSubmitClick(createEventForm)"
```
Comment (Getsov) [inline src/app/events/event-create/event-create.page.html]: The function can me named without the last word "Click", you already have "...Submit"

File: src/app/events/event-create/event-create.page.html [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+          ></ion-input>
+
+          <!-- event category selector -->
+          <div class="category-select-wrapper">
+            <ion-select
+              interface="popover"
+              placeholder="Вид на събитието"
+              class="region-select"
+              slot="end"
+              name="category"
```
Comment (Getsov) [inline src/app/events/event-create/event-create.page.html]: why "category"?

File: src/app/events/event-create/event-create.page.html [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+            [counter]="true"
+            maxlength="140"
+          ></ion-input>
+
+          <!-- event category selector -->
+          <div class="category-select-wrapper">
+            <ion-select
+              interface="popover"
+              placeholder="Вид на събитието"
+              class="region-select"
```
Comment (Getsov) [inline src/app/events/event-create/event-create.page.html]: region-select ?

File: src/app/events/event-create/event-create.page.html [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+              slot="end"
+              name="category"
+              #region
+              (ionChange)="onRegionChange($event.detail.value)"
```
Comment (Getsov) [inline src/app/events/event-create/event-create.page.html]: onRegionChange?

File: src/app/events/event-create/event-create.page.html (hunk 2 above)
Comment (AntonovAtanas) [reply src/app/events/event-create/event-create.page.html]: the backend will expect "category" as a property when creating the event

File: src/app/events/event-create/event-create.page.html (hunk 3 above)
Comment (AntonovAtanas) [reply src/app/events/event-create/event-create.page.html]: Changed with event-type-select

File: src/app/events/event-create/event-create.page.html (hunk 4 above)
Comment (AntonovAtanas) [reply src/app/events/event-create/event-create.page.html]: Deleted. Will refactor when i have the latest pull so i can populate Niki's Category enum

File: src/app/events/event-create/event-create.page.html (hunk 2 above)
Comment (AntonovAtanas) [reply src/app/events/event-create/event-create.page.html]: update: the event type will not go through the form, so the name is not needed at all.

**Review by Getsov — APPROVED** [review]

**Pull request merged**